experiment/autobumper/bumper/bumper_test.go

```diff
@@ -760,44 +807,104 @@ func TestCDToRootDir(t *testing.T) {
+		images            map[string]string
+		prefixes          []Prefix
+		expectedVersions1 map[string][]string
+		expectedVersions2 map[string][]string
+		err               bool
 	}{
 		{
-			name: "No images, no result, no error",
+			name:              "two prefixes being bumped with consistent tags",
+			prefixes:          []Prefix{prowPrefix, boskosPrefix},
+			images:            map[string]string{"gcr.io/k8s-prow/test:tag1": "newtag1", "gcr.io/k8s-prow/test2:tag1": "newtag1"},
+			err:               false,
+			expectedVersions1: map[string][]string{"newtag1": {"gcr.io/k8s-prow/test:tag1", "gcr.io/k8s-prow/test2:tag1"}},
+			expectedVersions2: map[string][]string{"newtag1": {"gcr.io/k8s-prow/test2:tag1", "gcr.io/k8s-prow/test:tag1"}},
 		},
 		{
-			name:        "One version is returned",
-			images:      map[string]string{"gcr.io/k8s-prow/deck": "v20200914-1ac05b0ca2"},
-			expectedTag: "v20200914-1ac05b0ca2",
+			name:     "two prefixes being bumped with inconsistent tags",
+			prefixes: []Prefix{prowPrefix, boskosPrefix},
+			images:   map[string]string{"gcr.io/k8s-prow/test:tag1": "newtag1", "gcr.io/k8s-prow/test:tag2": "newtag2"},
+			err:      true,
 		},
 		{
-			name:          "Multiple versions, error",
-			images:        map[string]string{"gcr.io/k8s-prow/deck": "v20200914-1ac05b0ca2", "gcr.io/k8s-prow/hook": "v20200915-1ac05", "gcr.io/k8s-prow/tide": "v20200915-1ac05"},
-			expectedError: `Expected a consistent version for all "gcr.io/k8s-prow/" images, but found multiple: [{v20200914-1ac05b0ca2 [gcr.io/k8s-prow/deck]} {v20200915-1ac05 [gcr.io/k8s-prow/hook gcr.io/k8s-prow/tide]}]`,
+			name:              "two prefixes being bumped with no bumps",
+			prefixes:          []Prefix{prowPrefix, boskosPrefix},
+			images:            map[string]string{},
+			err:               false,
+			expectedVersions1: map[string][]string{},
+		},
+		{
+			name:              "Prefix being bumped with inconsistent tags",
+			prefixes:          []Prefix{inconsistentPrefix},
+			images:            map[string]string{"inconsistent/test:tag1": "newtag1", "inconsistent/test2:tag2": "newtag2"},
+			err:               false,
+			expectedVersions1: map[string][]string{"newtag1": {"inconsistent/test:tag1"}, "newtag2": {"inconsistent/test2:tag2"}},
 		},
 	}
-
 	for _, tc := range testCases {
 		t.Run(tc.name, func(t *testing.T) {
-			var errMsg string
-			actualTag, err := getNewProwVersion(tc.images)
-			if err != nil {
-				errMsg = err.Error()
+			versions, err := getVersionsAndCheckConsistency(tc.prefixes, tc.images)
+			if tc.err && err == nil {
+				t.Errorf("expected error but did not get one")
 			}
-			if errMsg != tc.expectedError {
-				t.Fatalf("got error '%v', expected error %s", err, tc.expectedError)
+			if !tc.err && err != nil {
+				t.Errorf("expected no error, but got one: %v", err)
 			}
-			if err != nil {
-				return
+			if !reflect.DeepEqual(tc.expectedVersions1, versions) && !reflect.DeepEqual(tc.expectedVersions2, versions) {
```

Comment:
I think cmp.Diff should be easier here, if comparing the maps is an issue.

Reply:
DeepEqual seems to be working fine. The only issue is that sometimes the map returned is {a: [x, y]} and other times its {a: [y, x]} and I couldn't find a way around that other than checking both. cmp.Diff does not seem to help in this case. Any ideas on how to make the check easier?

Reply:
cmp.Diff has options you can pass to ignore order (I think "cmpopts.SortSlices" to compare slices ignoring order? https://godoc.org/github.com/google/go-cmp/cmp/cmpopts#SortSlices), which should be an easier comparison (especially if these tests ever get more complex).

Reply:
This is perfect! Thank you! :)